pkg/didcomm/protocol/didexchange/service_test.go

```diff
@@ -107,6 +107,30 @@ func TestService_Name(t *testing.T) {
 	require.Equal(t, DIDExchange, prov.Name())
 }
 
+func TestService_Handle_Panic(t *testing.T) {
+	dbstore, cleanup := store(t)
+	defer cleanup()
+
+	s := New(dbstore, &mockdid.MockDIDCreator{Doc: getMockDID()}, &mockProvider{})
+	actionCh := make(chan dispatcher.DIDCommAction)
+	close(actionCh)
```

Comment:
This doesn’t seem like a valid scenario. The client should call unregister first. If they don’t gracefully close things then panic will occur.

Reply:
ok, if it is expected behavior I do not mind